### PR TITLE
boards: arm: actinius_icarus: move sim selection to DT

### PR DESCRIPTION
**Pull request by gmarull**
Move sim selection to DT using a custom compatible and make use of
gpio_dt_spec.

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]

**Review by galak — DISMISSED** [review]
Should add something to the board docs about `sim-gpios`.

File: boards/arm/actinius_icarus/actinius_icarus_common.dts [hunk 1]
```diff
@@ -82,6 +82,12 @@
 		output-ohms = <10000000>;
 		full-ohms = <(10000000 + 4700000)>;
 	};
+
+	sim_select: sim-select {
+		compatible = "actinius,sim-select";
```
Comment (mbolivar-nordic) [inline boards/arm/actinius_icarus/actinius_icarus_common.dts:87]: Please read this and fix the compatible: https://docs.zephyrproject.org/latest/guides/dts/bindings.html#vendor-prefixes

Comment (gmarull) [reply boards/arm/actinius_icarus/actinius_icarus_common.dts]: thanks for the insight, removed `actinius,` prefix from the compatible.

**Comment by gmarull** [comment]
@galak ping

**Review by mbolivar-nordic — DISMISSED** [review]
Seems like a lot of duplicated code, but I don't have a better suggestion at this time

File: boards/arm/actinius_icarus/dts/bindings/sim-select.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The Icarus boards provide the user with 2 options regarding SIM usage for
+  LTE-M/NB-IoT communication. Either using the on-board eSIM or using an
+  external nano-SIM. The SIM can be selected using the SIM select pin given by
+  the sim-gpios property by setting it to 0 (eSIM) or to 1 (external nano-SIM).
+
+compatible: "sim-select"
```
Comment (carlocaione) [inline boards/arm/actinius_icarus/dts/bindings/sim-select.yaml:10]: should we use some kind of board-specific prefix here? like `icarus,sim-select` ?

Comment (carlocaione) [reply boards/arm/actinius_icarus/dts/bindings/sim-select.yaml:10]: ok, NVM, just saw https://github.com/zephyrproject-rtos/zephyr/pull/43817#pullrequestreview-916458049 . I thought we were going in the opposite direction.

**Review by mbolivar-nordic — APPROVED** [review]
New approach, same as the first. I don't have any suggestions for how to do it better, so OK by me even though there is a fair bit of duplication

**Review by carlescufi — APPROVED** [review]

**Comment by alextsam** [comment]
@gmarull @mbolivar-nordic @carlescufi 
This actually breaks the build with this board, as the SPM overrides the config with CONFIG_GPIO=n, thus with the missing GPIO dependency the build throws `undefined reference to `__device_dts_ord_7'`. Is there a way to force SPM to have CONFIG_GPIO=y? Otherwise we have to move board.c to only be included for the non-secure target.

**Comment by alextsam** [comment]
@gmarull You are correct, there was just an error log from the SPM before. And to be clear, I have absolutely no objection to using `DEVICE_DT_GET`. If anything, this is now forcing us to find a proper solution regarding GPIO dependency and SPM.

This particular pin (sim_select) can be skipped in the secure target build as it is not so important. However, there are cases of other pins (e.g. think of a charger enable pin) that would be good to have working in the secure target. So is there a way to force the dependency of the GPIO driver to be included in the SPM (other than overriding the whole SPM config with set spm_CONF_FILE in CMake)?
I am happy to make a separate issue about this if you think this is off-topic. I just thought it was easier to give context this way.

**Comment by alextsam** [comment]
@gmarull Thank you for your suggestion, but I am afraid my point did not come through:
As I mentioned in my messages, we could move the board init to happen only during non-secure but that doesn't solve the issue for (upcoming) pins that actually need to be set in secure, such as e.g. a charger enable pin. Also overriding completely the SPM config would not be a good solution. In other words, my question to you / @mbolivar-nordic / @carlescufi is whether there is a proper way to handle these dependencies of the board files in SPM.
Something similar to a config overlay `set(spm_OVERLAY_CONFIG ${CMAKE_CURRENT_LIST_DIR}/spm.conf)`.
As far as I know the above does not work from the board-level CMakeLists. And the <board>_defconfig is unfortunately being overridden by SPM's config.

**Pull request merged**